src/migrations/20230927172930-events-announced-index.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+exports.up = function (db, callback) {
+    db.runSql(
+        `
+        UPDATE events set announced = false where announced IS NULL;
+        ALTER TABLE events ALTER COLUMN announced SET NOT NULL;
+        ALTER TABLE events ALTER COLUMN announced SET DEFAULT false;
+        CREATE INDEX events_unannounced_idx ON events(announced) WHERE announced = false;
+        `,
+        callback(),
+    );
+};
+
+exports.down = function (db, callback) {
```

Comment:
I think a down function would be beneficial.

Reply:
I considered that but down migrations are not executed automatically, so if we want to make changes we're probably going to move forward rather than backward

Reply:
@sjaanus I know we're in a weird place with the down migrations and what we should do with them. I've never used them though. Are you using them? I don't remember any recommendations to users to consider down migrations either, our response has always been get latest version where we fixed the issue

Reply:
@sighphyre I have not used, but I know customers have backed down to previous versions and in some cases if we do not provide down migration, it may make their instance unusable.

For example, if we currently do not allow nulls and previously somehow they were entered, we would probably just have errors in logs, but in other cases, if we change something related to projects/strategies/segments, it might break UI.

Reply:
Oh yeah, if there are actually folks using them then it seems important

Reply:
> I know customers have backed down to previous versions and in some cases if we do not provide down migration, it may make their instance unusable.

Indeed, but even in these cases, there's no command for the customer to execute the down migration other than by hand. They'd have to either go one by one, or identify the one they need to execute. 

In such a scenario, the down migration does not update the migrations table (because we're not using the tool for this), and therefore the framework doing the up migration will still believe that the migration is applied and will never try to apply it again... this leaves the customer in a very weird situation. 

I know because we faced this same problem with our demo instance and we learned that down migrations should not be applied manually because they don't include the change in the migrations table (and they shouldn't! The right solution is to use the tool's command for doing it)